src/sys/unix.rs

```diff
@@ -2154,6 +2157,55 @@ impl crate::Socket {
             )
         }
     }
+
+    /// Get the value of the `TCP_CONGESTION` option for this socket.
+    ///
+    /// For more information about this option, see [`set_tcp_congestion`].
+    ///
+    /// [`set_tcp_congestion`]: Socket::set_tcp_congestion
+    #[cfg(all(feature = "all", any(target_os = "freebsd", target_os = "linux")))]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(all(feature = "all", any(target_os = "freebsd", target_os = "linux"))))
+    )]
+    pub fn tcp_congestion(&self) -> io::Result<Vec<u8>> {
+        let mut payload: MaybeUninit<[u8; TCP_CA_NAME_MAX]> = MaybeUninit::uninit();
+        let mut len = size_of::<[u8; TCP_CA_NAME_MAX]>() as libc::socklen_t;
+        syscall!(getsockopt(
+            self.as_raw(),
+            IPPROTO_TCP,
+            libc::TCP_CONGESTION,
+            payload.as_mut_ptr().cast(),
+            &mut len,
+        ))
+        .map(|_| {
+            let buf = unsafe { payload.assume_init() };
+            let name = buf.splitn(2, |num| *num == 0).next().unwrap().to_vec();
+            name
```

Comment:
I think the following will work.
```suggestion
            let buf = &buf[..len as usize];
            // TODO: use `MaybeUninit::slice_assume_init_ref` once stable.
            Ok(Some(unsafe { &*(buf as *const [_] as *const [u8]) }.into()))
```

Reply:
Here do you mean directly return the [u8; len](in vec format)? This may have more than one `\0` appreared in the Vec<u8> returned.

E.g., In Linux, tcp_ca name is `bbr` and the return value is `[98, 98, 114, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]`.

If this is intended, than I will do as you suggest.

Reply:
If the desired behavior is the user need to parse `[u8]` to some string themselves, then maybe not splitting the buffer is better.